src/vivarium_cluster_tools/psimulate/worker/vivarium_work_horse.py

```diff
@@ -203,3 +223,54 @@ def format_and_record_details(
                 df.insert(df.shape[1] - 1, col_name, val)
         finished_results_metadata[key] = val
     return finished_results_metadata
+
+
+def run(sim: SimulationContext, job_parameters: JobParameters):
+    backup_freq = job_parameters.backup_configuration["backup_freq"]
+    if backup_freq:
+        backup_dir = job_parameters.backup_configuration["backup_dir"]
+        backup_path = (backup_dir / str(get_current_job().id)).with_suffix(".pkl")
+        time_to_save = time() + backup_freq
+        while not sim.past_stop_time():
+            sim.step()
+            if time() >= time_to_save:
+                sim.write_backup(backup_path)
+                time_to_save = time() + backup_freq
+    else:
+        sim.run()
+
+
+def get_backup(job_parameters: JobParameters) -> Optional[SimulationContext]:
+    backup_dir = job_parameters.backup_configuration["backup_dir"]
+    if not backup_dir.exists() or not os.listdir(backup_dir):
+        # No sim backups to load
+        return None
+    metadata_path = job_parameters.backup_configuration["backup_metadata_path"]
+    try:
+        pickle_metadata = pd.read_csv(metadata_path)
+    except FileNotFoundError:
+        return None
+    query_conditions = f"input_draw == {job_parameters.input_draw} & random_seed == {job_parameters.random_seed}"
+
+    # Add branch parameter conditions to the query
+    for k, v in collapse_nested_dict(job_parameters.branch_configuration):
+        query_conditions += f' & `{k}` == "{v}"'
+
+    # Use the query method to find rows that match the lookup parameters
+    run_ids = pickle_metadata.query(query_conditions)["job_id"].to_list()
+    if run_ids:
+        filename = Path(backup_dir / run_ids[0]).with_suffix(".pkl")
```

Comment:
There should only ever be one row, right? Is there a way this could potentially be untrue? Might be worth raising if `len(run_ids) > 1`

Reply:
That should be the case. One thing I didn't really consider until now was what happens if we save backups, then restart and also save backups with the restarted run. I think VCT as written will overwrite the backup metadata for the *new* run instead of appending to the file, so there is only one job per branch. 

It seems to me that it would be better to append to the csv and then choose the most recently created matching file, do you agree?

Reply:
Can it instead replace the relevant row, rather than appending? Appending seems fine as well though.

Reply:
The actual backup files given by the backup metadata file aren't guaranteed to exist, so appending has the advantage of selecting the appropriate backup if, say, we do a restart but the job fails before creating a new backup, or the "most recent" backup gets deleted or corrupted.